code/controllers/subsystem.dm

```diff
@@ -98,7 +101,11 @@
 	SHOULD_NOT_OVERRIDE(TRUE)
 	set waitfor = 0
 	. = SS_SLEEPING
-	fire(resumed)
+	if(!wait_for_explosions)
+		fire(resumed)
+	else
+		if(!SSexplosions.is_exploding())
```

Comment:
should be a flag too imo something like MC_CONTROL_INTERRUPTED

Reply:
maybe even roll it into the fire priority system, allow pausing priority downwards or something. Or an independent system for multiple pause priorities.